### PR TITLE
Vanilla Template Update

### PR DESCRIPTION
**Pull request by oguzutku1745**
## Motivation

The old Vanilla template lacked a deploy function. This new template has deploy function.

## Test Plan

A deploy function has been added to both the frontend and workers.

image

## Related PRs

No functional change, this is a part of template updates. For the next.js template, check following [PR](https://github.com/AleoHQ/sdk/pull/879)

### CODE REVIEW

File: create-aleo-app/template-vanilla/worker.js [hunk 1]
```diff
@@ -1,47 +1,94 @@
...
+    PrivateKey,
+    initThreadPool,
+    AleoKeyProvider,
+    AleoNetworkClient,
+    NetworkRecordProvider,
+  } from "@aleohq/sdk";
+  
+  await initThreadPool();
+  
+  const hello_hello_program =
```
Comment (Pauan) [inline create-aleo-app/template-vanilla/worker.js]: What is the reason for changing the string?

File: create-aleo-app/template-vanilla/worker.js [hunk 2]
```diff
@@ -1,47 +1,94 @@
...
+  
+  async function localProgramExecution(program, aleoFunction, inputs) {
+      const programManager = new ProgramManager();
+  
+      // Create a temporary account for the execution of the program
+      const account = new Account();
+      programManager.setAccount(account);
+  
+      const executionResponse = await programManager.run(
+          hello_hello_program,
```
Comment (Pauan) [inline create-aleo-app/template-vanilla/worker.js]: Why this change?

File: create-aleo-app/template-vanilla/worker.js [hunk 3]
```diff
@@ -1,47 +1,94 @@
...
+  
+  onmessage = async function (e) {
+      if (e.data === "execute") {
+          const result = await localProgramExecution();
+          postMessage(result);
+      } else if (e.data === "key") {
+          const result = getPrivateKey();
+          postMessage(result);
+      } else if (e.data === "deploy") {
+        deployProgram(hello_hello_program).then(result => {
```
Comment (Pauan) [inline create-aleo-app/template-vanilla/worker.js]: It would be better to write this using `await` syntax:

```js
        try {
            const result = await deployProgram(hello_hello_program);
            postMessage(result);
        } catch (error) {
            postMessage({ error: error.message });
        }
```

**Review by Pauan — CHANGES_REQUESTED** [review]

File: create-aleo-app/template-vanilla/worker.js (hunk 1 above)
Comment (oguzutku1745) [reply create-aleo-app/template-vanilla/worker.js]: I changed the file name to be unique in order to test the deploy function. It looks like I forgot to change it back. Thanks for pointing this out; I'm updating it immediately!

File: create-aleo-app/template-vanilla/worker.js (hunk 2 above)
Comment (oguzutku1745) [reply create-aleo-app/template-vanilla/worker.js]: The function `executeOffline` has been renamed to "run" with [this commit](https://github.com/AleoHQ/sdk/commit/45ac2ad7645499e8efbf3e76e421e4233308c9dd#diff-d5500ae2c32e9410d2a9a44bb20b6daecae531f19c3a01e1e9c8e3c90e5f21bb). That is the reason for the change.

File: create-aleo-app/template-vanilla/worker.js (hunk 3 above)
Comment (oguzutku1745) [reply create-aleo-app/template-vanilla/worker.js]: Updating this now, thanks!

File: create-aleo-app/template-vanilla/worker.js (hunk 1 above)
Comment (Pauan) [reply create-aleo-app/template-vanilla/worker.js]: Thanks, but why did you change the way that the string is created?

File: create-aleo-app/template-vanilla/worker.js (hunk 2 above)
Comment (Pauan) [reply create-aleo-app/template-vanilla/worker.js]: What I mean is, the old code passed it in as arguments, but now you're hardcoding it.

File: create-aleo-app/template-vanilla/worker.js (hunk 1 above)
Comment (oguzutku1745) [reply create-aleo-app/template-vanilla/worker.js]: My bad, I tested the templates with both versions of this string. I thought there was no difference between the two and pushed with this one, but the older version is definitely clearer. Updated

File: create-aleo-app/template-vanilla/worker.js (hunk 2 above)
Comment (oguzutku1745) [reply create-aleo-app/template-vanilla/worker.js]: Yes, that was a mistake I didn't realize. It seems I was too focused on deploying functionality. I've corrected it now

**Review by Pauan — APPROVED** [review]
Looks good, thanks!
